### DISCUSSION TITLE
router.refresh()

### DISCUSSION THREAD

**Discussion by GabrielRCodes** [vercel/next.js#65121] [Help]
### Summary

I'm doing a Next.js project and I was doing some tests to find out how I could send information from the client to the server. I decided to use cookies to do this. However, this information is inside a loop, to be updated by a fetch every 3 seconds. The thing is, I managed to do a fetch, populate a state and update a cookie and at the end I used rounter.refresh. My question is, will this router.refresh() be "charged" to me somehow in vercel's plans? Like, what exactly does it spend? Because it's being used every 3 seconds to refresh the page only. Images below.

Set cookies + router refresh
![image](https://github.com/vercel/next.js/assets/76919248/6b10dd23-59e7-474f-aa2d-d64be0ca109d)

Getting the cookies on the server
![image](https://github.com/vercel/next.js/assets/76919248/68b8a280-686d-4aa0-b906-a63108070fba)

That worked for me, but I wanted to know if I get charged anything for it at vercel.

**Comment by hyjoong** [comment]
Your charges depend on where you performed the code to retrieve cookies from the server. Whether it's done in a serverless function, an edge function, or middleware will determine the pricing. You can find more details on pricing at the following links:

Edge function: [Vercel Edge Functions Pricing](https://vercel.com/docs/pricing/edge-functions)
Middleware: [Vercel Edge Middleware Pricing](https://vercel.com/docs/pricing/edge-middleware)
Serverless function: [Vercel Serverless Functions Pricing](https://vercel.com/docs/pricing/serverless-functions)

If you've deployed to Vercel, you can also monitor your charges in real-time on the Usage page.
